src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,200 @@
+                    newNode.previous = oldNode.previous;
+                    newNode.next = oldNode;
+                    Node<T> oldPrev = oldNode.previous;
+                    oldNode.previous = newNode;
+                    oldPrev.next = newNode;
+                    size++;
+                }
+            }
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null) {
+            return false;
+        }
+        for (T one: list) {
+            add(one);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findForIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
+        if (isIndexExist(index)) {
+            Node<T> requiredNode = findForIndex(index);
+            T oldElement = requiredNode.element;
+            requiredNode.element = value;
+            return oldElement;
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        if (isIndexExist(index)) {
+            Node<T> requiredNode = findForIndex(index);
+            if (index == 0 && size < 2) {
+                size--;
+                first = null;
+                last = null;
+                requiredNode.element = null;
+                return null;
+            }
+            if (index == 0) {
+                Node<T> oldNode = first;
+                first = findForIndex(index + 1);
+                oldNode.next = null;
+                size--;
+                return requiredNode != null ? requiredNode.element : null;
+            }
+            if (index >= 1 && index != size() - 1) {
+                Node<T> oldNext = findForIndex(index + 1);
+                Node<T> oldPrevious = findForIndex(index - 1);
+                oldPrevious.next = oldNext;
+                oldNext.previous = oldPrevious;
+                size--;
+                T requiredElement = requiredNode.element;
+                return requiredElement;
+            } else {
+                last = findForIndex(index - 1);
+                size--;
+                return requiredNode != null ? requiredNode.element : null;
+            }
+        }
         return null;
     }
 
     @Override
     public boolean remove(T t) {
+        if (size == 0) {
+            return false;
+        }
+        int removingNodeIndex = findForValue(t);
+        if (removingNodeIndex > - 1) {
+            remove(removingNodeIndex);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        if (last == null && first == null) {
+            return true;
+        }
         return false;
     }
+
+    private Node<T> findForIndex(int index) {
+        if (isIndexExist(index)) {
+            Node<T> requiredNode = first;
```

Comment:
try to optimize this check if index < size / 2 gor from first to middle of list, if not go from last ot middle